component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/AvroRecordBuilderFactoryProvider.java

```diff
@@ -46,14 +46,19 @@ public RecordBuilderFactory apply(final String containerId) {
             return new AvroRecordBuilderFactory(containerId);
         default:
             try {
-                ofNullable(Thread.currentThread().getContextClassLoader())
-                        .orElseGet(ClassLoader::getSystemClassLoader)
-                        .loadClass("org.codehaus.jackson.node.TextNode");
-                return new AvroRecordBuilderFactory(containerId);
+                final ClassLoader cl = ofNullable(Thread.currentThread().getContextClassLoader())
```

Comment:
While you have to rebase, extract the method that chose implementation :
```java
    @Override
    public RecordBuilderFactory apply(final String containerId) {
        switch (System.getProperty("talend.component.beam.record.factory.impl", "auto")) {
        case "memory":
        case "default":
            return new RecordBuilderFactoryImpl(containerId);
        case "avro":
            return new AvroRecordBuilderFactory(containerId);
        default:
            if (this.isAvroImpl()) {
                 return new AvroRecordBuilderFactory(containerId);
            }
            else {
                return new RecordBuilderFactoryImpl(containerId);
            }
        }
    }

protected boolean isAvroImpl() {
    try {
            final ClassLoader cl = ofNullable(Thread.currentThread().getContextClassLoader())
                        .orElseGet(ClassLoader::getSystemClassLoader);
            final Class<?> c1 = cl.loadClass("org.codehaus.jackson.node.TextNode");
            final Class<?> c2 = cl.loadClass("org.talend.sdk.component.runtime.beam.spi.record.AvroSchema");
            return c1.getClassLoader().equals(c2.getClassLoader());
     } catch (final ClassNotFoundException | NoClassDefFoundError cnfe) {
            log.info("component-runtime-beam is not available, skipping AvroRecordBuilderFactory ({}).",
                        getClass().getName());
                return false;
     }
}
```

Reply:
Done +1